src/Mollie/WC/Helper/ApplePayDirectHandler.php

```diff
@@ -0,0 +1,1180 @@
+                        '<a href="https://developer.apple.com/documentation/apple_pay_on_the_web/setting_up_your_server">',
+                        '</a>'
+                    );
+                    echo '</p></div>';
+                    return false;
+                }
+            );
+        }
+        add_action(
+            'woocommerce_after_add_to_cart_form',
+            function () {
+                $this->applePayDirectButton();
+            }
+        );
+        add_action(
+            'woocommerce_cart_totals_after_order_total',
+            function () {
+                $this->applePayDirectButton();
+            }
+        );
+        admin_url('admin-ajax.php');
+        $this->bootstrapAjaxRequest();
+    }
+
+    /**
+     * Checks if the server is HTTPS
+     *
+     * @return bool
+     */
+    private function isApplePayCompatible()
+    {
+        return !empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off';
+    }
+
+    /**
+     * Sets the appropriate data to send to ApplePay script
+     * Data differs between product page and cart page
+     *
+     * @return array|bool
+     */
+    public function applePayScriptData()
+    {
+        $base_location = wc_get_base_location();
+        $shopCountryCode = $base_location['country'];
+        $currencyCode = get_woocommerce_currency();
+        $totalLabel = get_bloginfo('name');
+        if (is_product()) {
+            $product = wc_get_product(get_the_id());
+            if (!$product) {
+                return false;
+            }
+            $isVariation = false;
+            if ($product->get_type() === 'variable') {
+                $isVariation = true;
+            }
+            $productNeedShipping = $this->checkIfNeedShipping($product);
+            $productId = get_the_id();
+            $productPrice = $product->get_price();
+
+            return [
+                'product' => [
+                    'needShipping' => $productNeedShipping,
+                    'id' => $productId,
+                    'price' => $productPrice,
+                    'isVariation' => $isVariation,
+                ],
+                'shop' => [
+                    'countryCode' => $shopCountryCode,
+                    'currencyCode' => $currencyCode,
+                    'totalLabel' => $totalLabel
+                ],
+                'ajaxUrl' => admin_url('admin-ajax.php')
+            ];
+        }
+        if (is_cart()) {
+            $cart = WC()->cart;
+
+            return [
+                'product' => [
+                    'needShipping' => $cart->needs_shipping(),
+                    'subtotal' => $cart->get_subtotal(),
+
+                ],
+                'shop' => [
+                    'countryCode' => $shopCountryCode,
+                    'currencyCode' => $currencyCode,
+                    'totalLabel' => $totalLabel
+                ],
+                'ajaxUrl' => admin_url('admin-ajax.php')
+            ];
+        }
+        return [];
```

Comment:
I would split this method into two other methods, and make this one the director. Sounds like something which can grow up in complexity.